### PR TITLE
Feature 68 error handler

### LINKED ISSUE #68: General error handler
As a user I'd like to have well described error messages if something goes wrong

### PR DESCRIPTION
**Pull request by EVP1989**
Resolves #68

### CODE REVIEW

File: src/components/contexts/error-handler.tsx [hunk 1]
```diff
@@ -20,17 +21,19 @@ export const ErrorContext = React.createContext<ErrorContextType>({
...
-   * Sends error message to sentry'
-   * TODO: Figure out proper way to log errors with sentry
+   * Sends error message to sentry
    *
    * @param message error message
    * @param err any error
    */
   const handleError = async (message: string, err?: any) => {
     setError(message);
+    console.log(err);
```
Comment (VottonDev) [inline src/components/contexts/error-handler.tsx:35]:
```suggestion
```

File: src/components/contexts/error-handler.tsx [hunk 2]
```diff
@@ -42,6 +45,7 @@ const ErrorHandler: React.FC = ({ children }) => {
       return;
     }
 
+    setErrorMessage(err.message);
     console.error(err);
```
Comment (VottonDev) [inline src/components/contexts/error-handler.tsx:49]:
```suggestion
```
Maybe could be removed as well? Has the setErrorMessage above already.

File: src/components/contexts/error-handler.tsx [hunk 3]
```diff
@@ -20,17 +21,19 @@ export const ErrorContext = React.createContext<ErrorContextType>({
...
-   * TODO: Figure out proper way to log errors with sentry
+   * Sends error message to sentry
    *
    * @param message error message
    * @param err any error
    */
   const handleError = async (message: string, err?: any) => {
     setError(message);
+    console.log(err);
+
```
Comment (VottonDev) [inline src/components/contexts/error-handler.tsx:36]:
```suggestion
  const handleError = async (message: string) => {
    setError(message);
    
```
Could remove the log as the setError message is already used?

File: src/localization/en.json [hunk 4]
```diff
@@ -9,11 +9,39 @@
...
   },
 
   "errorHandling": {
-    "title": "Error"
+    "title": "Error",
+    "dialog": {
+      "title": "Whoops! Something went wrong",
+      "reloadPage": "Try reloading the page.",
+      "unsavedContents": "Please note that unsaved content will not be saved when reloading. If you have unsaved content on this page, you can close this message and copy the content to the clipboard before reloading.",
+      "reportIssue": "If the problem persists, contact support@metatavu.fi by email. Please include technical details in to the email to speed up resolution of the problem.",
```
Comment (VottonDev) [inline src/localization/en.json:23]:
```suggestion
      "reportIssue": "If the problem persists, contact support@metatavu.fi by email. Please include technical details in to the email to speed up resolution of the problem.",
```
Personal preference, but "to" isn't really needed in the sentence.

File: src/components/contexts/error-handler.tsx [hunk 5]
```diff
@@ -52,6 +56,24 @@ const ErrorHandler: React.FC = ({ children }) => {
...
+   */
+  const getTime = () => {
+    return moment().format("DD.MM.YYYY HH:mm:ss");
+  };
+  
+  /**
+     * Returns current window URL
+     * 
+     * @returns current window URL
+     */
```
Comment (VottonDev) [inline src/components/contexts/error-handler.tsx:72]:
```suggestion
  /**
   * Returns current window URL
   * 
   * @returns current window URL
   */
```
Too many indentations in the doc

File: src/localization/en.json [hunk 6]
```diff
@@ -9,11 +9,39 @@
...
+      "technicalDetails": "technical details:",
```
Comment (VottonDev) [inline src/localization/en.json:24]:
```suggestion
      "technicalDetails": "Technical details:",
```
I'd probably make the first letter uppercase as not consistent at the moment with the other lines.
image

File: src/components/contexts/error-handler.tsx (hunk 2 above)
Comment (EVP1989) [reply src/components/contexts/error-handler.tsx:49]: setErrorMessage is for client to see but console.error could be useful for testing etc. It has the whole error message in it.

**Review by VottonDev — APPROVED** [review]
Looks good to me now.

**Pull request merged**